### PR TITLE
LPS-173648 Automation for LPS-171989 JS Polution Vulnerability

### PR DESCRIPTION
**Pull request by AkramYoussoufi**
- [Automation Request](https://liferay.atlassian.net/browse/LPS-173648)
- [Ticket](https://liferay.atlassian.net/browse/LPS-171989)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 1]
```diff
@@ -43,6 +43,47 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-171989"
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:46]: Add simple description for test

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 2]
```diff
@@ -43,6 +43,47 @@ definition {
...
+	@priority = 3
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:47]: @priority =4 should same with bug ticket

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 3]
```diff
@@ -43,6 +43,47 @@ definition {
...
+	test VerifyFramJSPolutionVulnerability {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:48]: Move to a new file IFrameWithXSS.testcase then rename to NavigateToSourceURLWithXSS

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 4]
```diff
@@ -43,6 +43,47 @@ definition {
...
+		task ("Given Configuration to the IFrame widget") {
+			Navigator.gotoSitePage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name");
+
+			Portlet.gotoPortletOptions(
+				portletName = "IFrame",
+				portletOption = "Configuration");
+
+			BlogsConfiguration.selectConfigurationIFrame();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:64]: IFrame.selectConfigurationFrame

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 5]
```diff
@@ -43,6 +43,47 @@ definition {
...
+
+			Type(
+				key_fieldLabel = "Source URL",
+				locator1 = "CustomFields#CUSTOM_INPUT_FIELD",
+				value1 = "http://${portalURL}:8080/web/guest/home");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:69]: InputField.editText

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 6]
```diff
@@ -43,6 +43,47 @@ definition {
...
+
+			IFrame.saveConfiguration();
+
+			Navigator.gotoSitePage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:75]: Iframe.closeFrame

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 7]
```diff
@@ -43,6 +43,47 @@ definition {
...
 	}
 
+	@description = "This is a use case for LPS-171989"
+	@priority = 3
+	test VerifyFramJSPolutionVulnerability {
+		var portalURL = JSONCompany.getPortalURL();
+
+		var portalURL = Regex.removeHTTP(url = ${portalURL});
+
+		var portalURL = Regex.removePortNumber(url = ${portalURL});
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:53]: `var portalURL = PropsUtil.get("portal.url");`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase (hunk 5 above)
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:69]: value1 = "https://www.liferay.com"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase [hunk 8]
```diff
@@ -43,6 +43,47 @@ definition {
...
+
+			IFrame.saveConfiguration();
+
+			Navigator.gotoSitePage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name");
+		}
+
+		task ("When Navigating to the IFrame location with url parameters in a new Tab") {
+			Open.openInTheNewTab(value1 = "http://${portalURL}:8080/web/test-site-name#constructor[prototype][comboBase]=data%3A%2Calert%281%29");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/misc/iframe/IFrame.testcase:79]: `Open.openInTheNewTab(value1 = "${portalURL}/web/test-site-name/test-page-name#constructor[prototype][comboBase]=data%3A%2Calert%281%29");`

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @AkramYoussoufi, Good job! Would you mind updating according to comments? Thanks

**Comment by AkramYoussoufi** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
@AkramYoussoufi, see changes https://github.com/liferay-echo/liferay-portal/pull/14497

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**